### PR TITLE
maint: update release job

### PR DESCRIPTION
**Pull request by vreynolds**
## Which problem is this PR solving?

- make next release a little nicer

## Short description of the changes

- use go 1.18 in CI
- create a **draft** release when publishing to GitHub

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -111,14 +111,11 @@ jobs:
                 command: |
                   echo "about to publish to tag ${CIRCLE_TAG}"
                   ls -l *
-                  ghr -t ${GITHUB_TOKEN} -u ${CIRCLE_PROJECT_USERNAME} -r ${CIRCLE_PROJECT_REPONAME} -c ${CIRCLE_SHA1} ${CIRCLE_TAG} ./artifacts/honeymarker-linux-386
-                  ghr -t ${GITHUB_TOKEN} -u ${CIRCLE_PROJECT_USERNAME} -r ${CIRCLE_PROJECT_REPONAME} -c ${CIRCLE_SHA1} ${CIRCLE_TAG} ./artifacts/honeymarker-linux-amd64
-                  ghr -t ${GITHUB_TOKEN} -u ${CIRCLE_PROJECT_USERNAME} -r ${CIRCLE_PROJECT_REPONAME} -c ${CIRCLE_SHA1} ${CIRCLE_TAG} ./artifacts/honeymarker-darwin-amd64
-                  ghr -t ${GITHUB_TOKEN} -u ${CIRCLE_PROJECT_USERNAME} -r ${CIRCLE_PROJECT_REPONAME} -c ${CIRCLE_SHA1} ${CIRCLE_TAG} ./artifacts/honeymarker-linux-arm64
+                  ghr -draft -n ${CIRCLE_TAG} -t ${GITHUB_TOKEN} -u ${CIRCLE_PROJECT_USERNAME} -r ${CIRCLE_PROJECT_REPONAME} -c ${CIRCLE_SHA1} ${CIRCLE_TAG} ./artifacts
```
Comment (robbkidd) [inline .circleci/config.yml:114]: upload them all at once

**Review by robbkidd — COMMENTED** [review]
```
?   	github.com/honeycombio/honeymarker	[no test files]
```

I guess it compiles under 1.18.

Should we update the `go 1.17` in `go.mod`, too?

**Review by JamieDanielson — APPROVED** [review]
perhaps a note that this updates #33 ?

**Review by robbkidd — APPROVED** [review]
marker approves

**Pull request merged**